tests/function-controller/pkg/subscription/subscription.go

```diff
@@ -0,0 +1,184 @@
+package subscription
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kymaeventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	"github.com/kyma-project/kyma/tests/function-controller/pkg/helpers"
+	"github.com/kyma-project/kyma/tests/function-controller/pkg/resource"
+	"github.com/kyma-project/kyma/tests/function-controller/pkg/shared"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	backOffJitter   = 0.1
+	backOffFactor   = 5.0
+	backOffSteps    = 10
+	backOffDuration = 4 * time.Second
+	eventType       = "sap.kyma.custom.something.order.created.v1"
+)
+
+type Subscription struct {
+	resCli      *resource.Resource
+	name        string
+	namespace   string
+	waitTimeout time.Duration
+	log         *logrus.Entry
+	verbose     bool
+}
+
+func New(name string, c shared.Container) *Subscription {
+	gvr := schema.GroupVersionResource{
+		Group:    "eventing.kyma-project.io",
+		Version:  "v1alpha1",
+		Resource: "subscriptions",
+	}
+
+	return &Subscription{
+		resCli:      resource.New(c.DynamicCli, gvr, c.Namespace, c.Log, c.Verbose),
+		name:        name,
+		namespace:   c.Namespace,
+		waitTimeout: c.WaitTimeout,
+		log:         c.Log,
+		verbose:     c.Verbose,
+	}
+}
+
+func (s *Subscription) Create(sinkUrl *url.URL) (subscription *kymaeventingv1alpha1.Subscription, err error) {
+	eventSource := &kymaeventingv1alpha1.Filter{
+		Type:     "exact",
+		Property: "source",
+		Value:    "",
```

Comment:
Is it equivalent of a wildcard?

Reply:
Not really, for NATS this should be blank for now. The source(or application name in Kyma context) is part of event-type for now. This source will be re-evaluated when BEB comes into picture.